packages/react/collection/src/Collection.tsx

```diff
@@ -2,11 +2,14 @@ import React from 'react';
 import { useComposedRefs } from '@radix-ui/react-compose-refs';
 import { Slot } from '@radix-ui/react-slot';
 
-type SlotProps = React.ComponentProps<typeof Slot>;
+// We have resorted to returning slots directly rather than exposing primitives that can then
+// be slotted like `<CollectionItem as={Slot}>…</CollectionItem>`. This is because we were having
+// issues with types because there are generics at play and it's not statically analyzable because
+// the components are returned by a dynamic function.
```

Comment:
```suggestion
// We have resorted to returning slots directly rather than exposing primitives that can then
// be slotted like `<CollectionItem as={Slot}>…</CollectionItem>`.
// This is because we encountered issues with generic types, static analysis 
// and the dynamic function used to return the components.
```

Reply:
> This is because we encountered issues with generic types that cannot be statically analyised
> due to being created dynamically via `createCollection`.